--- ggadget/qt/Makefile.in.orig	2009-02-27 20:32:39.000000000 -0300
+++ ggadget/qt/Makefile.in	2009-02-27 20:33:29.000000000 -0300
@@ -392,7 +392,7 @@
 @GGL_BUILD_LIBGGADGET_QT_TRUE@			  $(PREDEFINED_MACROS)
 
 @GGL_BUILD_LIBGGADGET_QT_TRUE@libggadget_qt@GGL_EPOCH@_la_LIBADD = \
-@GGL_BUILD_LIBGGADGET_QT_TRUE@			  $(QT4_LIBS) \
+@GGL_BUILD_LIBGGADGET_QT_TRUE@			  $(QT4_LIBS) -lX11 -lXext \
 @GGL_BUILD_LIBGGADGET_QT_TRUE@			  $(top_builddir)/ggadget/libggadget@GGL_EPOCH@.la \
 @GGL_BUILD_LIBGGADGET_QT_TRUE@			  $(top_builddir)/ggadget/xdg/libggadget-xdg@GGL_EPOCH@.la
 
